presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -351,6 +396,9 @@ private static void alterTableColumnTypes(String tableName)
         executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN int_to_bigint int_to_bigint bigint", tableName));
         executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN bigint_to_varchar bigint_to_varchar string", tableName));
         executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN float_to_double float_to_double double", tableName));
+        executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN row_to_row row_to_row struct<tinyint_to_smallint:smallint,tinyint_to_int:int,tinyint_to_bigint:bigint,smallint_to_int:int,smallint_to_bigint:bigint,int_to_bigint:bigint,bigint_to_varchar:string,float_to_double:double>", tableName));
+        executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN list_to_list list_to_list array<struct<tinyint_to_smallint:smallint,tinyint_to_int:int,tinyint_to_bigint:bigint,smallint_to_int:int,smallint_to_bigint:bigint,int_to_bigint:bigint,bigint_to_varchar:string,float_to_double:double>>", tableName));
+        executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN map_to_map map_to_map map<int,struct<tinyint_to_smallint:smallint,tinyint_to_int:int,tinyint_to_bigint:bigint,smallint_to_int:int,smallint_to_bigint:bigint,int_to_bigint:bigint,bigint_to_varchar:string,float_to_double:double>>", tableName));
```

Comment:
ditto here. (See my comment in `tableDefinitionBuilder` in AbstractTestHiveClient)